### DISCUSSION TITLE
fzcd plugin

### DISCUSSION THREAD

**Discussion by leo-arch** [leo-arch/clifm#89] [General]
Hi @Docbroke. First of all, thanks for this new plugin. 

* I've been playing around with it and, reading the article about the dangers of `ls(1)` in scripts, I came up with slightly different version aimed to avoid breaking file names with new line embedded chars:

```sh
find . -type d -print0 2> /dev/null | fzf --read0 ...
```
Only thing is, `-print0` parameter for `find` isn't POSIX.

* This is your plugin modified to match other plugins interface:

```sh
#!/bin/sh

# Change directory plugin for CliFM

# Find and change directory using find and fzf
# Author: Docbroke <contact email here>
# License: GPL3

DIR="$(find . -type d -print0 2> /dev/null | \
fzf --read0 --prompt "Change DIR: " \
--reverse --height 15 \
--bind "right:accept,tab:accept" \
--color=fg+:reverse,bg+:236,prompt:6,pointer:2,marker:2:bold,spinner:6:bold)"

if [ -n "$DIR" ]; then
    printf "%s\n" "$DIR" > "$CLIFM_BUS"
fi

exit 0
```

* It would be nice if we can make it work for parent directories as well. I tried replacing `.` by `/`, but the result isn't satisfactory at all. Maybe some flag to alter the pluing's behavior, dunno.

* I linked `fzcd` to `_` (underscore). We need a default command name for this plugin.

EDIT: `right:accept` should be removed from the parameters passed to `fzf`: it interferes when you try to edit the query string moving backwards/forwards with the arrow keys. I guess just TAB and Enter are fine.

**Comment by Docbroke** [comment]
yes `right:accept` is not right choice.
What is wrong with replacing `.` with `/`? It appears to be working with me.

**Reply by leo-arch** [reply]
Yeap, I made a quick test and seemed to me that search results weren't as expected when using root instead of CWD. But it seems to work fine indeed.

**Reply by leo-arch** [reply]
So, if you're ok with the new changes, we can update this plugin.

**Reply by leo-arch** [reply]
I forgot, I generally perform an existence check (in this case of `fzf`) before actually running `fzf`.

**Reply by Docbroke** [reply]
Yes, I am ok.

**Reply by leo-arch** [reply]
Great.

**Reply by leo-arch** [reply]
Btw, I'm using an environment variable to globally set `fzf` window's height in CliFM's plugins: `CLIFM_FZF_HEIGHT`. In this way we can have a more consistent style for plugins.

EDIT: I'm even thinking of supporting more variables for other fzf setttings, e.g. `CLIFM_FZF_COLORS`, and so on.

**Reply by Docbroke** [reply]
fzf already have `FZF_DEFAULT_OPTS`, it may interfere with these options.

**Comment by Docbroke** [comment]
I think plugins using `fzf` with `preview` needs more height, otherwise 10-15 lines are good enough.

**Reply by leo-arch** [reply]
Maybe.
